### PR TITLE
HOV-31 | Adds Stepper component

### PR DESCRIPTION
**Pull request by Raalzz**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: lib/src/components/Stepper/Stepper.tsx [hunk 1]
```diff
@@ -0,0 +1,87 @@
+import React, { Children, cloneElement, ForwardRefRenderFunction } from "react";
+import { Flex } from "src/components/Flex";
+
+import { TStepperProps } from "./stepper.types";
```
Comment (pruthvi2103) [inline lib/src/components/Stepper/Stepper.tsx:4]: change to `IStepperProps`

File: lib/src/components/Stepper/stepper.types.ts [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  dividerProps?: DividerProps;
+};
+
+export type TStepperStepProps = JSX.IntrinsicElements["div"] &
+  Omit<TStepperProps, "activeStep" | "children"> & {
+    children?: ReactNode;
+    ref?: MutableRefObject<HTMLDivElement | null>;
+  };
+
+export type TStepperTheme = [
```
Comment (pruthvi2103) [inline lib/src/components/Stepper/stepper.types.ts:34]: why use `T` here? `I` meaning interface should be used

**Review by pruthvi2103 — CHANGES_REQUESTED** [review]
Minor variable related changes

**Comment by saurabhsutar192** [comment]
DIvider LGTM

**Comment by pruthvi2103** [comment]
@Raalzz also i see one code-smell being introduced

File: lib/src/components/Stepper/stepper.types.ts (hunk 2 above)
Comment (Raalzz) [reply lib/src/components/Stepper/stepper.types.ts:34]: My bad, will Update the same

**Comment by Raalzz** [comment]
> @Raalzz also i see one code-smell being introduced

Yaa checked it

> @Raalzz also i see one code-smell being introduced

Its asking me to remove `.reduce` 🥲

File: lib/src/components/Stepper/stepper.types.ts (hunk 2 above)
Comment (pruthvi2103) [reply lib/src/components/Stepper/stepper.types.ts:34]: ahhh okok we are using type here, until now we use `I` regardless to seperate variables from types

Comment (pruthvi2103) [reply lib/src/components/Stepper/stepper.types.ts:34]: lets change it to`I` it think having both `T` and `I` will cause confusions in future

**Comment by pruthvi2103** [comment]
sad, we will look at codesmell later then

**Review by pruthvi2103 — APPROVED** [review]
LGTM

**Pull request merged**